scavenger-agent-java/src/integrationTest/java/integrationTest/support/AgentRunner.java

```diff
@@ -0,0 +1,85 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.Callable;
+import java.util.stream.Collectors;
+
+public class AgentRunner implements Callable<String> {
+    private final String javaPath;
+    private final String classpath;
+    private final String agentPath;
+
+    private String configFilePath;
+    private String cmdLineArgs;
+
+    public AgentRunner(String javaPath, String classpath, String agentPath) {
+        this.javaPath = javaPath;
+        this.classpath = classpath;
+        this.agentPath = agentPath;
+    }
+
+    public void setConfigFilePath(String configFilePath) {
+        this.configFilePath = configFilePath;
+    }
+
+    public void setConfig(Properties properties) {
+        cmdLineArgs = properties.entrySet().stream()
+            .map(entry -> entry.getKey() + "=" + entry.getValue())
+            .collect(Collectors.joining(";"));
+    }
+
+    @Override
+    public String call() throws IOException, InterruptedException, RuntimeException {
+        List<String> command = buildCommand();
+        Process process = new ProcessBuilder().command(command).redirectErrorStream(true).start();
+        String output = collectProcessOutput(process.getInputStream());
+        int exitCode = process.waitFor();
+        if (exitCode != 0) {
+            throw new RuntimeException(String.format("Could not execute '%s': %s%nExit code=%d", command, output, exitCode));
+        }
+        return output;
+    }
+
+    private List<String> buildCommand() {
+        String cp = classpath.endsWith(":") ? classpath.substring(0, classpath.length() - 2) : classpath;
+
+        List<String> command = new ArrayList<>();
+        command.add(javaPath);
+        if (cmdLineArgs != null) {
+            command.add("-javaagent:" + agentPath + "=" + cmdLineArgs);
+        } else {
+            command.add("-javaagent:" + agentPath);
+        }
+        command.add("-cp");
+        command.add(cp);
+        command.add("-Djava.util.logging.config.file=src/integrationTest/resources/logging.properties");
+        command.add("-Duser.language=en");
+        command.add("-Duser.country=US");
+        if (configFilePath == null) {
+            command.add("-Dscavenger.configuration=src/integrationTest/resources/scavenger.conf");
+        } else if (!configFilePath.isEmpty()) {
+            command.add("-Dscavenger.configuration=" + configFilePath);
+        }
+        command.add("sample.app.SampleApp");
+        System.out.printf("%nLaunching SampleApp with the command: %s%n%n", command);
+        return command;
+    }
+
+    private static String collectProcessOutput(InputStream inputStream) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        String line;
+        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
```

Comment:
this reader is not closed.
You should use 
```try(BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream)) {```

Reply:
fixed in [81f8510](https://github.com/naver/scavenger/pull/71/commits/81f851082322f23c4f2d31cafaa411bddc58e8f7).